pkg/controller/pxc/controller.go

```diff
@@ -180,7 +181,13 @@ func (r *ReconcilePerconaXtraDBCluster) Reconcile(request reconcile.Request) (re
 		return reconcile.Result{}, err
 	}
 
-	err = r.updatePod(statefulset.NewNode(o), o.Spec.PXC, o)
+	operatorPod, err := r.operatorPod(o)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "get operator deployment")
+	}
+
+	epInit := statefulset.EntrypointInitContainer(operatorPod.Spec.Containers[0].Image)
+	err = r.updatePod(statefulset.NewNode(o), o.Spec.PXC, o, []corev1.Container{epInit})
```

Comment:
Does it make any sense to update the init-container? I mean it should run once while pod is creating. Or I'm wrong? If so why PorxySQL's init-containers are not being updated?

Reply:
This func does not update init container, I pass init container to this func because it has already had logic when init container was put to pod https://github.com/percona/percona-xtradb-cluster-operator/blob/K8SPXC-256/pkg/controller/pxc/upgrade.go#L95